cumulus_library/statistics/psm.py

```diff
@@ -0,0 +1,247 @@
+# Module for generating Propensity Score matching cohorts
+
+import numpy as np
+import pandas
+import toml
+
+from psmpy import PsmPy
+
+
+import json
+from pathlib import PosixPath
+from dataclasses import dataclass
+
+from cumulus_library.cli import StudyBuilder
+from cumulus_library.databases import DatabaseCursor
+from cumulus_library.base_table_builder import BaseTableBuilder
+from cumulus_library.template_sql.templates import (
+    get_ctas_query_from_df,
+    get_drop_view_table,
+)
+from cumulus_library.template_sql.statistics.psm_templates import (
+    get_distinct_ids,
+    get_create_covariate_table,
+)
+
+
+@dataclass
+class PsmConfig:
+    """Provides expected values for PSM execution
+
+    These values should be read in from a toml configuration file.
+    See tests/test_data/psm/psm_config.toml for an example with details about
+    the expected values for these fields.
+    """
+
+    classification_json: str
+    pos_source_table: str
+    neg_source_table: str
+    target_table: str
+    primary_ref: str
+    count_ref: str
+    count_table: str
+    dependent_variable: str
+    pos_sample_size: int
+    neg_sample_size: int
+    join_cols_by_table: dict[str, dict]
+    seed: int
+
+
+class PsmBuilder(BaseTableBuilder):
+    """TableBuilder for creating PSM tables"""
+
+    display_text = "Building PSM tables..."
+
+    def __init__(self, toml_config_path: str):
+        """Loads PSM job details from a PSM configuration file"""
+        with open(toml_config_path, encoding="UTF-8") as file:
+            toml_config = toml.load(file)
+        self.config = PsmConfig(
+            classification_json=f"{PosixPath(toml_config_path).parent}/{toml_config['classification_json']}",
```

Comment:
nit: Some fields here are directly looked up and some are optional with a default. The user experience if some of these fields are missing might be kinda obscure? They'd get some KeyError I assume. May be worth wrapping this with an `except KeyError` and give a nice little user error. But could also be a future improvement.